pkg/identity/identity.go

```diff
@@ -232,12 +240,43 @@ func LookupReservedIdentityByLabels(lbls labels.Labels) *Identity {
 			return nil
 
 		case lbl.Source == labels.LabelSourceReserved:
-			// If it contains the reserved, local host identity, return it with
-			// the new list of labels. This is to ensure the local node retains
-			// this identity regardless of label changes.
 			id := GetReservedID(lbl.Key)
-			if id == ReservedIdentityHost {
-				identity := NewIdentity(ReservedIdentityHost, lbls)
+			switch {
```

Comment:
I believe that case is not handled through reserved identities since the identity is generated on-the-fly as a CIDR identity (just with extra labels than a regular CIDR identity, it has all the cidr labels but also `reserved:kube-apiserver`.

Reply:
Yeah basically this function returns nil and the allocation falls through to the "local identity generation" case.